api/v1/submissions/PKPSubmissionController.php

```diff
@@ -664,8 +664,34 @@ public function add(Request $illuminateRequest): JsonResponse
                     : $submitAsUserGroup->permitMetadataEdit
             );
 
+        $isSubmittingAsProductionEditor = $submitAsUserGroup->getLocalizedData('name') === __('default.groups.name.productionEditor');
```

Comment:
Not sure if this is the best way to check for a specific role. Open to suggestions

Reply:
I think the best way would be to check the access to the submission stage through the user group stage table

Reply:
Forgot to mention that OPS might require different treatment - in 3.5 stage ID has only value `5` there, submission stage was removed.